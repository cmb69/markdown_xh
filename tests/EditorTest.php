<?php

namespace Markdown;

use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;
use Plib\FakeRequest;
use Plib\View;

class EditorTest extends TestCase
{
    private View $view;

    protected function setUp(): void
    {
        $this->view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["markdown"]);
    }

    private function sut(): Editor
    {
        return new Editor("./", "./plugins/markdown/", $this->view);
    }

    public function testInitRendersHjs(): void
    {
        $request = new FakeRequest();
        $response = $this->sut()->init([], false, $request);
        Approvals::verifyHtml($response->hjs());
    }

    public function testInitRenderBjs(): void
    {
        $request = new FakeRequest();
        $response = $this->sut()->init([], false, $request);
        Approvals::verifyHtml($response->bjs());
    }

    public function testReplaceReturnCorrectJS(): void
    {
        $request = new FakeRequest();
        $response = $this->sut()->replace("#the-editor", false, $request);
        $this->assertSame('markdown.initById("#the-editor")', $response);
    }

    public function testIncludesOnlyOnce(): void
    {
        $request = new FakeRequest();
        $sut = $this->sut();
        $sut->include($request);
        $response = $sut->include($request);
        $this->assertNull($response->hjs());
        $this->assertNull($response->bjs());
    }
}
