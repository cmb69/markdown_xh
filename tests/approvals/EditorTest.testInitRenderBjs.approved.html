
<dialog class="markdown_modal" data-url="/?&amp;filebrowser=editorbrowser&amp;editor=markdown&amp;prefix=.%2F&amp;type=TYPE">
  <p class="markdown_buttons">
    <button type="button">close</button>
  </p>
  <div>
    <iframe src="about:blank"></iframe>
  <div>
</dialog>
<script src="./plugins/markdown/markdown/markdown.js"></script>
<script>init_tinyMDE()</script>
